legal-api/src/legal_api/services/filings/validations/restoration.py

```diff
@@ -68,15 +68,16 @@ def validate(business: Business, restoration: Dict) -> Optional[Error]:
     return None
 
 
-def validate_expiry_date(filing: Dict, restoration_type: str) -> list:
+def validate_expiry_date(business: Business, filing: Dict, restoration_type: str) -> list:
     """Validate expiry date."""
-    # Between 1 month and 2 years in the future for limited restoration
-    # Between 1 month and 3 years in the future for limited restoration extension
     msg = []
     expiry_date_path = '/filing/restoration/expiry'
+
     if expiry_date := get_date(filing, expiry_date_path):
-        now = LegislationDatetime.now().date()
-        max_expiry_years = 2 if restoration_type == 'limitedRestoration' else 3
+        max_expiry_years = 2
+        now = LegislationDatetime.as_legislation_timezone(business.restoration_expiry_date).date()
+        if restoration_type == 'limitedRestoration': now = LegislationDatetime.now().date()
```

Comment:
`restoration_expiry_date` will be null while filing a `limitedRestoration`.
```suggestion
        now = LegislationDatetime.now().date()
        if restoration_type == 'limitedRestorationExtension': 
            now = LegislationDatetime.as_legislation_timezone(business.restoration_expiry_date).date()
```